services/autoscaling/src/simcore_service_autoscaling/modules/auto_scaling_mode_base.py

```diff
@@ -0,0 +1,99 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from fastapi import FastAPI
+from models_library.docker import DockerLabelKey
+from models_library.generated_models.docker_rest_api import Node as DockerNode
+from servicelib.logging_utils import LogLevelInt
+
+from ..models import AssociatedInstance, EC2InstanceData, EC2InstanceType, Resources
+
+
+@dataclass
+class BaseAutoscaling(ABC):  # pragma: no cover
```

Comment:
One day I would like to pick your mind on the advantages on `ABC` when it comes to python. I still have my doubts about the return of investment on this but I guess that with mypy-like tools this has a bigger RoI

Reply:
I guess I could also use a Protocol somehow, but I needed to have an object instance so I went ABC